salt/dex/dex.yaml

```diff
@@ -118,6 +118,19 @@ spec:
       tolerations:
       - key: "CriticalAddonsOnly"
```

Comment:
Should we also update the tolerations to all dex to run on the masters? which will give us some extra copies in smaller clusters? With a smaller test / POC cluster, we can easily have just a single worker node - giving us just 1 dex pod. If we allow it masters, we get the 3 we request.

Reply:
That's a good question.

If we want to do that we have to start label the master nodes accordingly. Upstream usually sets `node-role.kubernetes.io/master=true`.

By doing that we could write a new toleration rule like:

```yaml
tolerations:
  - key: node-role.kubernetes.io/master
    value: "true"
    effect: PreferNoSchedule
```

Reply:
I put this together which sets the taints and labels etc: https://github.com/kubic-project/salt/pull/388

Reply:
Lets move this discussion over to https://github.com/kubic-project/salt/pull/388 - I've added it there, as well as correctly tainting the masters.